### DISCUSSION TITLE
Need help with config regarding launching applications on windows

### DISCUSSION THREAD

**Discussion by rianahmed985** [wezterm/wezterm#4274] [Q&A]
I've been reading the docs on how to launch programs from the specific doc section but for some reason i literally just can't get it to work I've used the first command

-- Spawn a fish shell in login mode
config.default_prog = { '"C:\Program Files\PowerShell\7\pwsh.exe"', '-l' }

but this leads to an error

and the other command i've tried is this

wezterm start -- "C:\Program Files\PowerShell\7\pwsh.exe"

but it doesn't work any help would be appreciated

**Comment by juupas** [comment]
I have also an issue launching Powershell from Wezterm, I've set WSL Ubuntu as default and that works, as well as the "Bash" in my menu. When I try to launch Powershell the tab just closes and I get no log output on what's going on, pretty hard to debug.

This is my config:
```
-- Pull in the wezterm API
local wezterm = require("wezterm")

-- This table will hold the configuration.
local config = {}

-- In newer versions of wezterm, use the config_builder which will
-- help provide clearer error messages
if wezterm.config_builder then
        config = wezterm.config_builder()
end

-- This is where you actually apply your config choices

-- For example, changing the color scheme:
config.color_scheme = "Solarized (dark) (terminal.sexy)"

config.default_domain = 'WSL:Ubuntu'

config.launch_menu = {
  {
    label = 'Powershell',
    args = { 'C:/Program Files/PowerShell/7/pwsh.exe' },
  },
  {
    -- Optional label to show in the launcher. If omitted, a label
    -- is derived from the `args`
    label = 'Bash',
    -- The argument array to spawn.  If omitted the default program
    -- will be used as described in the documentation above
    args = { 'bash', '-l' },

    -- You can specify an alternative current working directory;
    -- if you don't specify one then a default based on the OSC 7
    -- escape sequence will be used (see the Shell Integration
    -- docs), falling back to the home directory.
    -- cwd = "/some/path"

    -- You can override environment variables just for this command
    -- by setting this here.  It has the same semantics as the main
    -- set_environment_variables configuration option described above
    -- set_environment_variables = { FOO = "bar" },
  },
}

-- and finally, return the configuration to wezterm
return config
```

**Reply by johnchenkzy** [reply]
I have this problem too, and I only can get errorinfo by snapshot.here is some infomation,but i have no idea about this...
```
execvpe pwsh.exe failed 2
```

**Reply by johnchenkzy** [reply]
I find a config for this problem!This work for me, maybe you can try it. 
```lua
  {
    label = 'Powershell',
    domain = { DomainName = "local" },
    args = { 'C:/Program Files/PowerShell/7/pwsh.exe' },
  },
```
Because in wsl domain maybe changed,set to "local" to resovle it. Can See Doc [Here](https://wezfurlong.org/wezterm/config/lua/SpawnCommand.html)

**Comment by juupas** [comment]
For your case, just use forward slashes like this:
`wezterm start -- "C:/Program Files/PowerShell/7/pwsh.exe"`

**Comment by Joker1718** [comment]
Use these in windows. Change the pathing of your default launcher/program, if needed.

**Put the config into your %USERPROFILE%\.config\wezterm\wezterm.lua**
```

local wezterm = require("wezterm")
local config = wezterm.config_builder()

config.default_prog = { "C:\\Program Files\\PowerShell\\7\\pwsh.exe" }
return config
```
